### ISSUE TITLE
Explorer don't refresh file change status automatically

### ISSUE THREAD

**Issue by abgnwl** [microsoft/vscode#226615]
Type: <b>Bug</b>

I'm using remote-ssh environment. When I modify a remote file,  the Explorer and file status are not refreshing automatically. There is no 'M' symbol in Explorer. I have to manually click Refresh in Source Control to see git changes. (manually refreshing the Explorer has no effect)

I have tried to set "git.autorefresh": true, but it doesn't work.

VS Code version: Code 1.85.2 (8b3775030ed1a69b13e4f4c628c612102e30a681, 2024-01-18T06:40:10.514Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 5.4.241-1-tlinux4-0017.1

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz (8 x 3000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.78GB (16.22GB free)|
|Process Argv|--crash-reporter-id 887f76a1-42a5-4491-a127-ae98a413d02b|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: cloud-dev|
|OS|Linux x64 5.4.241-1-tlinux4-0017.1|
|CPUs|AMD EPYC 7K62 48-Core Processor (32 x 2595)|
|Memory (System)|62.55GB (55.34GB free)|
|VM|0%|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-ssh|ms-|0.107.1
vscode-language-pack-zh-hans|MS-|1.92.2024081409
docker|vsc|1.0.0
yaml|vsc|1.0.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
945dj816:31013170
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
impr_priority:31102340
refactort:31108082
pythonrstrctxt:31112756
flightc:31119335
wkspc-onlycs-t:31111718
wkspc-ranged-t:31123313
fje88620:31121564

```

</details>

**Comment by abgnwl** [comment]
Sometimes VSCode will refresh the file status (but I don't know what operation triggered it), and the git OUTPUT will display `[info]>git status - z - all`.

However, in most cases (including saving or opening files), there is no automatically refresh and no `git status - z - all` in git OUTPUT.

**Comment by lszomoru** [comment]
@abgnwl, does setting the `typescript.tsserver.experimental.useVsCodeWatcher` setting to `false` help resolve the issue?

**Comment by abgnwl** [comment]
This command is not working.

While searching `typescript.tsserver.experimental.useVsCodeWatcher`, I found that I added `**` to `Files: Watcher Exclude`. (following https://github.com/microsoft/vscode/issues/3998)
After deleting '**', Explorer can automatically refresh file change status.

Thank you for providing the solution, it inspired me.

![image](https://github.com/user-attachments/assets/95c86e27-86da-4e3d-9aab-250e2b80426d)
